src/app/Table.elm

```diff
@@ -1,29 +1,95 @@
-module Table exposing (view, init, Msg)
+module Table exposing (view, init, initWithPlayersAndCup)
+
+{-|
+
+   This module handles the table of players,
+   and everything relating to that.
+
+-}
 
 import Html exposing (..)
+import Html.Events exposing (..)
+import Types exposing (PlayerID)
+import TableMsg exposing (Msg(..))
+import Dice
 
 
-{-
+type alias AppMsg =
+    Types.Msg
 
-   This module handles the view rendering for the table of players
-   and everything relating to that.
 
--}
+type alias PlayerName =
+    String
+
+
+type alias DiceCount =
+    Int
+
+
+type alias TablePlayer =
+    { id : PlayerID
+    , name : PlayerName
+    , diceCount : DiceCount
+    }
 
 
-type Model
-    = Model
+type alias TablePlayers =
+    List TablePlayer
 
 
-type Msg
-    = Msg
+type alias Model =
+    { tablePlayers : TablePlayers
+    , myCup : Dice.Cup
+    }
 
 
 init : Model
 init =
-    Model
+    { tablePlayers = noPlayers
+    , myCup = Dice.emptyCup
+    }
+
+
+initWithPlayersAndCup : List ( Int, String, Int ) -> List Int -> Model
+initWithPlayersAndCup tuples cup =
+    let
+        tupleToPlayer ( id, name, diceCount ) =
+            { id = id, name = name, diceCount = diceCount }
+    in
+        { tablePlayers = List.map tupleToPlayer tuples
+        , myCup = Dice.cupFromIntList cup
+        }
+
+
+noPlayers : TablePlayers
+noPlayers =
+    []
+
+
+testPlayers : TablePlayers
+testPlayers =
+    [ TablePlayer 1 "Joe" 6, TablePlayer 2 "Garrett" 5 ]
+
+
+view : (Msg -> AppMsg) -> Model -> Html AppMsg
+view privateMsg { tablePlayers } =
+    div []
+        [ button [ onClick <| privateMsg CallLiar ] [ text "Liar!" ]
```

Comment:
Oh... turns out it *is* possible (haha probably the way the actual TEA structure worked in the first place, I guess :headslap:) ;-)